File: java/BubbleSort.java
class BubbleSort 
{
	void bubbleSort(int arr[])
	{
		int n = arr.length;
		for(int i = 0;i<n-1;i++)
		{
			for(int j = 0;j<n-i-1;j++)
			{
				if(arr[j] > arr[j+1])
				{
					int temp = arr[j];
					arr[j] = arr[j+1];
					arr[j+1] = temp;

				}
			}
		}
	}

	void printArray(int arr[])
	{
		System.out.println("Printing Array : ");
		for (int i=0;i<arr.length;i++)
		{
			System.out.print(arr[i]+" ");
		}
	}

	public static void main(String[] args) {
		BubbleSort obj = new BubbleSort();
		int arr[] = {12,02,11,45,87,2,1,4,87,88,10,-6};
		obj.bubbleSort(arr);
		obj.printArray(arr);


	}
}

File: c/bubble_sort.c
#include <stdio.h>
#include <stdlib.h>

int main(){
int N;
printf("Enter the number of elements in the array\n");
scanf("%d",&N);
int *arr=malloc(N*sizeof(*arr));
printf("\nEnter the elemnts of the array\n");
for(int i=0;i<N;i++){
  scanf("%d",&arr[i]);
}
printf("\nOriginal Array\n");
for(int i=0;i<N;i++){
  printf("%d ",arr[i]);
}
int temp;
for(int i=0;i<N-1;i++){
  for(int j=0;j<N-i-1;j++){
    if(arr[j]>arr[j+1]){
      temp=arr[j];
      arr[j]=arr[j+1];
      arr[j+1]=temp;
    }
  }
}
printf("\nSorted Array\n");
for(int i=0;i<N;i++){
  printf("%d ",arr[i]);
}
}